destination/destination.go

```diff
@@ -0,0 +1,213 @@
+package destination
+
+//go:generate paramgen -output=paramgen_dest.go DestinationConfig
+//go:generate mockgen -source=destination.go -package=mock -destination=mock/client_mock.go -mock_names=weaviateClient=WeaviateClient . weaviateClient
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+
+	"github.com/conduitio-labs/conduit-connector-weaviate/config"
+	"github.com/conduitio-labs/conduit-connector-weaviate/destination/weaviate"
+	sdk "github.com/conduitio/conduit-connector-sdk"
+	"github.com/google/uuid"
+)
+
+var (
+	metadataClass = "weaviate.class"
+)
+
+type weaviateClient interface {
+	Open(weaviate.Config) error
+
+	Insert(context.Context, *weaviate.Object) error
+	Update(context.Context, *weaviate.Object) error
+	Delete(context.Context, *weaviate.Object) error
+}
+
+type Destination struct {
+	sdk.UnimplementedDestination
+
+	config DestinationConfig
+	client weaviateClient
+}
+
+type ModuleHeader struct {
+	// Name of the header configuring a module (e.g. `X-OpenAI-Api-Key`)
+	Name string `json:"name"`
+	// Value for header given in `moduleHeader.name`.
+	Value string `json:"value"`
+}
+
+func (m ModuleHeader) IsValid() bool {
+	return (m.Name == "" && m.Value == "") ||
+		(m.Name != "" && m.Value != "")
+}
+
+type DestinationConfig struct {
+	config.Config
+	//TODO: better naming for this value __sL__
+	// Vectorizers which can be configured client side
+	// mostly require an API key only.
+	// However, OpenAI can also be configured with an organization
+	// via the X-OpenAI-Organization header.
+
+	ModuleHeader ModuleHeader `json:"moduleHeader"`
+	// Whether a UUID for records should be automatically generated.
+	// The generated UUIDs are MD5 sums of record keys.
+	GenerateUUID bool `json:"generateUUID"`
+}
+
+func New() sdk.Destination {
+	return NewWithClient(&weaviate.Client{})
+}
+
+func NewWithClient(client weaviateClient) sdk.Destination {
+	return sdk.DestinationWithMiddleware(
+		&Destination{client: client},
+		sdk.DefaultDestinationMiddleware()...,
+	)
+}
+
+func (d *Destination) Parameters() map[string]sdk.Parameter {
+	return d.config.Parameters()
+}
+
+func (d *Destination) Configure(ctx context.Context, cfg map[string]string) error {
+	sdk.Logger(ctx).Info().Msg("Configuring Destination...")
+	err := sdk.Util.ParseConfig(cfg, &d.config)
+	if err != nil {
+		return fmt.Errorf("invalid config: %w", err)
+	}
+
+	if !d.config.ModuleHeader.IsValid() {
+		return errors.New("invalid module configuration")
```

Comment:
this might be a bit clearer
```suggestion
		return errors.New("invalid moduleHeader configuration")
```